### ISSUE TITLE
[IOS] RadioListTile and CheckboxListTile does not have correct semantics rect in landscape mode

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#82952]
Internal: b/183130084

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  @override
  State createState() => _MyAppState();
}

enum SingingCharacter { lafayette, jefferson }

class _MyAppState extends State<MyApp> {
  bool useSemanticsDebugger = false;

  SingingCharacter _character = SingingCharacter.lafayette;

  // Widget buildBody(BuildContext context) {
  //   return Column(
  //     children: <Widget>[
  //       RadioListTile<SingingCharacter>(
  //         title: const Text('Lafayette'),
  //         value: SingingCharacter.lafayette,
  //         groupValue: _character,
  //         onChanged: (SingingCharacter? value) {
  //           setState(() {
  //             _character = value!;
  //           });
  //         },
  //       ),
  //       RadioListTile<SingingCharacter>(
  //         title: const Text('Thomas Jefferson'),
  //         value: SingingCharacter.jefferson,
  //         groupValue: _character,
  //         onChanged: (SingingCharacter? value) {
  //           setState(() {
  //             _character = value!;
  //           });
  //         },
  //       ),
  //     ],
  //   );
  // }

  Widget buildBody(BuildContext context) {
    return Column(
      children: <Widget>[
        CheckboxListTile(
          title: const Text('Lafayette'),
          value: _character == SingingCharacter.lafayette,
          onChanged: (bool? value) {
            setState(() {
              _character = value!
                  ? SingingCharacter.lafayette
                  : SingingCharacter.jefferson;
            });
          },
        ),
        CheckboxListTile(
          title: const Text('Thomas Jefferson'),
          value: _character == SingingCharacter.jefferson,
          onChanged: (bool? value) {
            setState(() {
              _character = value!
                  ? SingingCharacter.jefferson
                  : SingingCharacter.lafayette;
            });
          },
        ),
      ],
    );
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        showSemanticsDebugger: useSemanticsDebugger,
        theme: ThemeData(primaryColor: Colors.blue),
        home: Scaffold(
          appBar: AppBar(
            title: Text('Semantics Test'),
          ),
          body: buildBody(context),
        ));
  }
}

```
1. on ios rotate the screen into landscape mode
2. turn on voiceover
3. swipe to focus the list tile

expect: the accessibility focus rect correct surround the list tile
actual: the accessibility focus rect drawn out of place.

**Comment by chunhtai** [comment]
We create a native UISwitch(aka FlutterSwitchSemanticsObject) for every checkbox or radio box, we attaches it to the accessibility tree but not the actual view, I think this is where iOS is not happy with. When the screen load or rotate, I believe there is some internal logic try to snapshot the current view hierarchy. I got this message when ever the item is focus first after screen orientation changed

[Snapshotting] Drawing a view (0x11f1747a0, FlutterSwitchSemanticsObject) that has not been rendered at least once requires afterScreenUpdates:YES.

I tried different trick to make it believe it has changed but none of them work. There are two things i can try.
1. Attaches the FlutterSwitchSemanticsObject to the view hierarchy, so that the iOS may be happier.
2. Flip the relationship of FlutterSwitchSemanticsObject and SemanticsObject. Right now the FlutterSwitchSemanticsObject wraps the SemanticsObject, and the iOS interact with FlutterSwitchSemanticsObject. The FlutterSwitchSemanticsObject delegates its method call to SemanticsObject. If make SemanticsObject wraps the FlutterSwitchSemanticsObject, the iOS do not interact with FlutterSwitchSemanticsObject directly and it may not care whether the FlutterSwitchSemanticsObject is rendered or not.

**Comment by chunhtai** [comment]
Ah yes, attach FlutterSwitchSemanticsObject to the view works. Now, I just need to figure out how to hide it correctly...
